### PR TITLE
Fix lakectl upload from standard input

### PR DESCRIPTION
**Pull request by guy-har**
Still need to fix the test

### CODE REVIEW

File: cmd/lakectl/cmd/fs_upload.go [hunk 1]
```diff
@@ -32,13 +32,16 @@ var fsUploadCmd = &cobra.Command{
 
 		ctx, stop := signal.NotifyContext(ctx, os.Interrupt, os.Kill)
 		defer stop()
-
-		stat, err := os.Stat(source)
-		if err != nil {
-			Die("failed to stat source", 1)
+		ignoreRecessive := true // Ignore recursive if source is the standard input or a file (not a directory)
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/fs_upload.go:35]: "recessive"? 🧬 
Possibly spelling correction gone wrong.

Comment (arielshaqed) [inline cmd/lakectl/cmd/fs_upload.go:35]: This should probably be a function: it already needs it's own documentation, might as well give it a name and a godoc.

File: cmd/lakectl/cmd/fs_upload.go [hunk 2]
```diff
@@ -32,13 +32,16 @@ var fsUploadCmd = &cobra.Command{
...
+		if source != StdinFileName {
+			stat, err := os.Stat(source)
+			if err != nil {
+				Die("failed to stat source", 1)
+			}
+			ignoreRecessive = !stat.IsDir()
 		}
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/fs_upload.go:43]: Do we need all this?  The changes seem functionally identical to
```go
		if source == StdinFileName {
			recursive = false
		}
```
and then checking for a directory.

File: esti/lakectl_test.go [hunk 3]
```diff
@@ -113,7 +113,13 @@ func TestLakectlPreSignUpload(t *testing.T) {
 
 	filePath := "ro_1k.1"
 	vars["FILE_PATH"] = filePath
-	RunCmdAndVerifySuccessWithFile(t, Lakectl()+" fs upload -s files/ro_1k lakefs://"+repoName+"/"+mainBranch+"/"+filePath+" --pre-sign", false, "lakectl_fs_upload", vars)
+	t.Run("upload from file", func(t *testing.T) {
+		RunCmdAndVerifySuccessWithFile(t, Lakectl()+" fs upload -s files/ro_1k lakefs://"+repoName+"/"+mainBranch+"/"+filePath+" --pre-sign", false, "lakectl_fs_upload", vars)
+	})
+	t.Run("upload from stdin", func(t *testing.T) {
+		RunCmdAndVerifySuccessWithFile(t, "echo sample | "+Lakectl()+" fs upload -s - lakefs://"+repoName+"/"+mainBranch+"/"+filePath+" --pre-sign", false, "lakectl_fs_upload", vars)
+		RunCmdAndVerifySuccess(t, "echo sample | "+Lakectl()+" fs upload -s - lakefs://"+repoName+"/"+mainBranch+"/"+filePath+" --pre-sign", false, "sample", vars)
```
Comment (arielshaqed) [inline esti/lakectl_test.go:121]: What's the difference between these two `verify`s?

**Review by arielshaqed — APPROVED** [review]
Thanks!

I think we can either move the business logic out to another func, or try to simplify it - proposal inside.

But nothing blocking here, and this fixes a real bug.

File: cmd/lakectl/cmd/fs_upload.go (hunk 1 above)
Comment (guy-har) [reply cmd/lakectl/cmd/fs_upload.go:35]: Too much trust on the auto-correct 😞 
Thanks

File: esti/lakectl_test.go (hunk 3 above)
Comment (guy-har) [reply esti/lakectl_test.go:121]: The first should be removed, the second should be fixed, still working on it

File: cmd/lakectl/cmd/fs_upload.go (hunk 2 above)
Comment (guy-har) [reply cmd/lakectl/cmd/fs_upload.go:43]: But in order to check for a directory I need to stat the file and I can't stat if source != StdinFileName 
So I will need this `if` again.

File: cmd/lakectl/cmd/fs_upload.go (hunk 1 above)
Comment (arielshaqed) [reply cmd/lakectl/cmd/fs_upload.go:35]: Still recessive 🙀 .

File: cmd/lakectl/cmd/fs_upload.go (hunk 2 above)
Comment (arielshaqed) [reply cmd/lakectl/cmd/fs_upload.go:43]: Not sure where we need the stat id `!recursive`.

**Review by arielshaqed — APPROVED** [review]
Thanks! LTTM, but I think we can check whether or not we are recursive in  a more straightforward way. Consider extracting that business logic to a function, then we can see how much early returns can help us.

**Comment by guy-har** [comment]
@arielshaqed I'm not sure what happened here, didn't push any changes yet or ask for a re-review. sorry that you went over it again. I will request a re-review once I handle all your comments and add a test that passes

File: cmd/lakectl/cmd/fs_upload.go (hunk 2 above)
Comment (guy-har) [reply cmd/lakectl/cmd/fs_upload.go]: @arielshaqed PTAL

**Pull request merged**